### DISCUSSION TITLE
How can I replace each occurrance of a sequence with another?

### DISCUSSION THREAD

**Discussion by SpacewaIker** [helix-editor/helix#12082] [Q&A]
In this case, I have a text file with tab characters for indentation that I want to replace with four spaces. I can use `indent-style 4`, but then there's no command (as far as I'm aware) to fix the indentation because there's no formatter for a text file.

What I could do with vim, is use substitution to substitute every tab character with four spaces: `:%s/\t/    /g`. Generally, I'd use something like `%s` in helix, but if I do `%s\t`, that will indeed select all tab characters, but pressing `c` to change that and put my four spaces will put _one_ cursor _per group of tabs_, not one per tab. So in the lines where I have two tabs, there will only be four spaces, not eight.

But this isn't just related to indenting, anytime I have two selections made with `s` that touch, then `c` will only put one insert cursor. For instance, if I have `()()()` and I select the whole thing, then press `s\(\)`, and `c[]`, I'd expect this to replace all occurrences of `()` with `[]`, but in reality, it inserts exactly one `[]`.

How can I achieve this?

**Comment by TornaxO7** [comment]
> What I could do with vim, is use substitution to substitute every tab character with four spaces: :%s/\t/    /g. Generally, I'd use something like %s in helix, but if I do %s\t, that will indeed select all tab characters, but pressing c to change that and put my four spaces will put one cursor per group of tabs, not one per tab. So in the lines where I have two tabs, there will only be four spaces, not eight

I think the trick here is to use the `replace` and/or the `insert` keybinding and navigate to the left and right instead of just using `c`. Although you do have to think a bit more about it which steps you'd like to do, it should be doable with that.
Nevertheless, I see that this may be a bit confusing. Maybe someone else has a better idea.

**Reply by kirawi** [reply]
This is a limitation of our system. Kakoune does it as expected (they visually collapse into one cursor, but there are still actually multiple cursors) but nobody has put in the work on implementing it yet.

**Comment by chtenb** [comment]
You can press `i` for insert mode, add however many spaces you want, exit insert mode, then press `d` for deleting the tab characters.

**Comment by mweihs** [comment]
If you are on Linux, you could pipe to `expand -t4` for tabs->4 spaces and `tr \(\) []` for the ().
